src/main/java/io/fabric8/maven/docker/config/handler/compose/DockerComposeServiceWrapper.java

```diff
@@ -182,11 +182,22 @@ NetworkConfig getNetworkConfig() {
             NetworkConfig ret = new NetworkConfig(NetworkConfig.Mode.custom, custom);
             Object aliases = toJoin.get(custom);
             if (aliases != null) {
-                if (!(aliases instanceof List)) {
-                    throwIllegalArgumentException("'networks:' Aliases must be given as a list of string");
-                }
-                for (String alias : (List<String>) aliases) {
-                    ret.addAlias(alias);
+                if (aliases instanceof List) {
+                    for (String alias : (List<String>) aliases) {
+                        ret.addAlias(alias);
+                    }
+                } else if (aliases instanceof LinkedHashMap) {
+                    LinkedHashMap<String, ArrayList<String>> map = (LinkedHashMap<String, ArrayList<String>>) aliases;
```

Comment:
`Map<String,List<String>> map` because using the implementation classes as type parameters really causes a tight coupling and you really don't need this.

Reply:
You are right again